ee/api/test/test_event_definition.py

```diff
@@ -242,3 +243,99 @@ def test_event_definition_no_duplicate_tags(self):
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json()["count"], 2)  # installed_app, rated_app
+        self.assertEqual(response.json()["results"][0]["name"], "installed_app")
+        self.assertEqual(response.json()["results"][1]["name"], "rated_app")
+
+        excluded_ids_str = f'["{str(ids[0])}"]'
+        response = self.client.get(
+            f'/api/projects/@current/event_definitions/?search=app&{urllib.parse.urlencode({"excluded_ids": excluded_ids_str})}'
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json()["count"], 1)
+        self.assertEqual(response.json()["results"][0]["id"], str(ids[1]))
+        self.assertEqual(response.json()["results"][0]["name"], "installed_app")
+
+        response = self.client.get(
+            f'/api/projects/@current/event_definitions/?search=app&{urllib.parse.urlencode({"excluded_ids": []})}'
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json()["count"], 2)  # installed_app, rated_app
+        self.assertEqual(response.json()["results"][0]["name"], "installed_app")
+        self.assertEqual(response.json()["results"][1]["name"], "rated_app")
+
+    def test_included_ids_overrides_excluded_ids_filter(self):
+        super(LicenseManager, cast(LicenseManager, License.objects)).create(
+            plan="enterprise", valid_until=timezone.datetime(2500, 1, 19, 3, 14, 7)
+        )
+        # rated_app, installed_app
+        rated_app_event = EnterpriseEventDefinition.objects.create(team=self.team, name="rated_app")
+        installed_app_event = EnterpriseEventDefinition.objects.create(team=self.team, name="installed_app")
+        ids = [rated_app_event.id, installed_app_event.id]
+
+        response = self.client.get("/api/projects/@current/event_definitions/?search=app")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json()["count"], 2)  # installed_app, rated_app
+        self.assertEqual(response.json()["results"][0]["name"], "installed_app")
+        self.assertEqual(response.json()["results"][1]["name"], "rated_app")
+
+        ids_str = f'["{str(ids[0])}"]'
+        response = self.client.get(
+            f'/api/projects/@current/event_definitions/?search=app&{urllib.parse.urlencode({"excluded_ids": ids_str, "included_ids": ids_str})}'
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.json()["count"], 2)
+        self.assertEqual(response.json()["results"][0]["id"], str(ids[0]))
+        self.assertEqual(response.json()["results"][0]["name"], "rated_app")
+
+        response = self.client.get(
+            f'/api/projects/@current/event_definitions/?search=app&{urllib.parse.urlencode({"excluded_ids": [], "included_ids": []})}'
```

Comment:
Judging by all these tests, these parameters aren't exactly filters, are they? Seems more like sorting. No matter what's in included or excluded, the results is always all event definitions, sorted differently?

Reply:
That's exactly right for `included_ids`, where all event definitions are returned with definitions in included ids at the top of the list. `excluded_ids` acts as a more generic filter that will omit any definitions with those ids from the returned list.

Reply:
But I agree some better documentation will make this less confusing. Perhaps I could also rename this to `included_ids_first` to better represent what we're doing here?